Animation.cs

```diff
@@ -0,0 +1,71 @@
+﻿using Microsoft.Xna.Framework.Graphics;
+using SixLabors.ImageSharp;
+using System;
+using System.Collections.Generic;
+using Microsoft.Xna.Framework;
+
+namespace Rampastring.XNAUI;
+
+class Frame
+{
+    public Texture2D Texture { get; set; }
+    public int Delay { get; set; }
+}
+
+public class Animation
+{
+    private List<Frame> Frames;
+    private int currentFrameId = 0;
+    private int currentDelay = 0;
+    private int totalElapsedTime = 0;
+
+    public Texture2D CurrentFrame { get; private set; }
+    public int Height { get; }
+    public int Width { get; }
+
+    public Animation(string value) 
+    {
+        Frames = new List<Frame>();
+        
+        var gif = AssetLoader.LoadAnimation(value);
+        currentDelay = gif.Frames[0].Metadata.GetGifMetadata().FrameDelay * 10;
+
+        Height = gif.Height;
+        Width = gif.Width;
+
+        // ImageSharp doesn't return last gif frame
+        int len = gif.Frames.Count - 1;
+
+        for (int i = 0; i < len; i++)
+        {
+            // ImageSharp returns not milliseconds, but decisecond
+            var delay = gif.Frames[0].Metadata.GetGifMetadata().FrameDelay * 10;
+            var currentFrame = gif.Frames.ExportFrame(0);
+
+            Frames.Add(new Frame { Texture = AssetLoader.TextureFromImage(currentFrame), Delay = delay });
+        }
+
+        CurrentFrame = Frames[0].Texture;
```

Comment:
I don't like how having this in the constructor makes this class hardwired to support GIFs only. It's restrictive if there is ever a need to implement support for other animated formats, like APNG.

Instead you could have a `Animation.FromGIF` static method that parses and returns an animation from GIF.

Reply:
I don't think a static method can be a good solution. As an alternative, extracting GIF animation initialization code to the specific method and setting a switch-case to catch file extension to load exact animation would be better.